### PR TITLE
NVIC/Cortex-M0: IPR and SCS_SHPR access must be words!

### PR DESCRIPTION
**Pull request by UweBonnes**

### CODE REVIEW

**Comment by karlp** [comment]
this has whitespace problems at least.  will look at the rest later  Fixes https://github.com/libopencm3/libopencm3/issues/957

File: lib/cm3/nvic.c [hunk 1]
```diff
@@ -161,6 +163,23 @@ void nvic_set_priority(uint8_t irqn, uint8_t priority)
 		/* Device specific interrupts */
 		NVIC_IPR(irqn) = priority;
 	}
+#else
+        volatile uint32_t *block;
+        unsigned int index;
+	if (irqn < NVIC_IRQ_COUNT) {
```
Comment (karlp) [inline lib/cm3/nvic.c]: is this relying on it being unsigned and that the negatives will appear to be higher?

File: lib/cm3/nvic.c [hunk 2]
```diff
@@ -143,6 +143,7 @@ uint8_t nvic_get_irq_enabled(uint8_t irqn)
  *
  * There are 4 priority levels only, given by the upper two bits of the
  * priority byte, as required by ARM standards. No grouping available.
+ * Only word access allowed!
```
Comment (karlp) [inline lib/cm3/nvic.c]: given that people are using the api for this, this comment is meaningless for them.

File: lib/cm3/nvic.c (hunk 1 above)
Comment (UweBonnes) [reply lib/cm3/nvic.c]: Same code as original code

File: lib/cm3/nvic.c (hunk 2 above)
Comment (UweBonnes) [reply lib/cm3/nvic.c]: Unclear wording. Is with regard to the Cortex-M registers, not the API access.

**Comment by UweBonnes** [comment]
I am happy with #1072 to. Should I rework my patch or do you regard #1072 better,

**Comment by karlp** [comment]
re whitespace, you get "make stylecheck" and try not to make things worse.  I'll look at both of them tonight.

**Comment by karlp** [comment]
merged the first implentation in d8579dde thanks!

**Pull request closed without merging**